### PR TITLE
Document and test all supported Python versions

### PR DESCRIPTION
**Pull request by jdufresne**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +21,7 @@ matrix:
         - TOXENV=py35
 
 install:
-  - pip install -U tox
+  - pip install tox
```
Comment (gsnedders) [inline .travis.yml]: I'd strongly prefer we kept using `-U`; Travis has a bunch of Python modules installed out of the box, some of which are absolutely ancient (i.e., several years old). Given I've been bitten by things not updating them before and ending up with dependencies not being met before (le sigh pip and and its lack of any real dependency resolution) and things messily breaking, I'd much rather just guarantee we were using the latest version of tox and all its dependencies. Or just hardcode specific versions of all.

**Comment by jdufresne** [comment]
> I'd strongly prefer we kept using `-U`

Makes sense. I have reverted that line.

**Comment by gsnedders** [comment]
@jdufresne sorry for somewhat dropping the ball on this (I got ill, then travelling); see my latest commit directly on your branch for a somewhat needless perf optimization (but hey, saving ~10 seconds per build job isn't bad); thanks for this, though!

**Comment by gsnedders** [comment]
Also sorry for overusing semicolons in the above comment. :)

**Pull request merged**